shell/common/gin_helper/dictionary.h

```diff
@@ -110,6 +126,22 @@ class Dictionary : public gin::Dictionary {
         .ToChecked();
   }
 
+  template <typename K, typename V>
+  bool SetDeprecated(const K& key,
+                     const V& val,
+                     const std::string& message,
+                     v8::PropertyAttribute attribute = v8::None) {
+    auto context = isolate()->GetCurrentContext();
+
+    v8::Local<v8::Value> getter = gin::ConvertToV8(
+        isolate(), base::BindRepeating(&DeprecatedValueProvider<V>, message,
+                                       isolate(), val));
+    v8::PropertyDescriptor prop_desc(getter, v8::Undefined(isolate()));
+    v8::Maybe<bool> maybe = GetHandle()->DefineProperty(
+        context, gin::StringToV8(isolate(), key), prop_desc);
```

Comment:
Maybe a good opportunity to use `v8::Object::SetLazyDataProperty` ([source definition](https://source.chromium.org/chromium/chromium/src/+/main:v8/include/v8-object.h;l=365-378;drc=0d0cbbeedf5831fe32e4823c0e5a854983c64ed8)). This would run the getter once, then replace the property with the resulting value. That way repeatedly accessing the property would only emit the deprecation warning once.

Untested code:
```suggestion
    v8::Maybe<bool> maybe = GetHandle()->SetLazyDataProperty(
        context, gin::StringToV8(isolate(), key), prop_desc);
```